### ISSUE TITLE
🤖 and 🪵 are rendered incorrectly it would seem

### ISSUE THREAD

**Issue by Jasper-Bekkers** [dfrg/swash#26]
Repro case

It looks like both of these render partially (I haven't tried many more, however 🔥 from the documentation seems to render correctly). The robot-man seems to have his mouth cut off, and the log seems to have the top cut off.

Since this is my very first time diving into this library, I may well be doing something wrong. I've attached the repro of what I'm trying to do.

I've also attached the rendered image below;

```rust
use swash::scale::ScaleContext;
use swash::scale::StrikeWith;
use swash::scale::*;
use zeno::Vector;

fn main() {
    use swash::FontRef;

    let font_path = "C:/Windows/Fonts/seguiemj.ttf";

    let font_data = std::fs::read(font_path).ok().unwrap();
    let font = FontRef::from_index(&font_data, 0).unwrap();

    let mut context = ScaleContext::new();
    let mut scaler = context.builder(font).hint(true).build();
    let glyph_id = font.charmap().map('🤖');
    let image = Render::new(&[
        Source::ColorBitmap(StrikeWith::ExactSize),
        Source::ColorOutline(1),
        Source::Outline,
    ])
    .format(zeno::Format::Subpixel)
    .offset(Vector::new(0.0, 0.0))
    .render(&mut scaler, glyph_id)
    .unwrap();

    dbg!(image.placement);

    let img =
        ::image::RgbaImage::from_raw(image.placement.width, image.placement.height, image.data)
            .unwrap();
    img.save_with_format("stuff.png", ::image::ImageFormat::Png);
}
```

![stuff](https://user-images.githubusercontent.com/49594/159183908-70ab325c-8132-49e2-833e-21b1902c7042.png)

**Comment by rice7th** [comment]
Hmm, weird, i tried running your code and there was no image generated.. (image.data was just empty)

**Comment by Jasper-Bekkers** [comment]
That is indeed quite odd; I'm running with the following Cargo.toml

```
[package]
name = "swash-test"
version = "0.1.0"
edition = "2021"

# See more keys and their definitions at https://doc.rust-lang.org/cargo/reference/manifest.html

[dependencies]
swash = "0.1.4"
zeno = "0.2"
image = "0.24"
```

I'm also running on Windows (as you can see by the font path), maybe that makes a difference? I just re-ran the code hoping it was potentially a version issue, but I'm getting a clean `stuff.png` image generated which looks identical to the image I've attached here.

**Comment by Jasper-Bekkers** [comment]
Digging in a bit, it's starting to look like some of the scaling code is incorrectly calculating the size of the image for colored outlines.

https://github.com/dfrg/swash/blob/d3f8bbfa5f96688ab94278db6f85e00ba383e3f8/src/scale/mod.rs#L914-L920

I don't think it's correct to take the very first `placement` as the final image's total size, I have some local modifications that calculate the sizes and offsets beforehand but it's not quite correct yet (and to be a bit faster would need some minor changes to `zeno` to make some functions public). I'll play around with this some more to see if I can figure out the correct logic, though having a spec handy to compare to would be nice, so if anyone has a relevant link for me I'd love to take a look.

**Comment by Jasper-Bekkers** [comment]
Checking it a bit further, it seems like either implementation is out of spec for ignoring the clipbox; https://learn.microsoft.com/en-us/typography/opentype/spec/colr#metrics-and-boundedness-of-color-glyphs-using-version-1-formats

**Comment by Jasper-Bekkers** [comment]
> In any case, I’ll dig into those particular glyphs and see if I can find out what’s going on. In the worst case, we take the union of the bounding box of all layers and that should give us appropriate dimensions.

The PR that I filed does this already (I hope correctly)

**Comment by Jasper-Bekkers** [comment]
Fixed by #30
